commands/checkout.go

```diff
@@ -119,9 +119,19 @@ func transformCheckoutArgs(args *Args) error {
 				newBranchName = fmt.Sprintf("%s-%s", pullRequest.Head.Repo.Owner.Login, pullRequest.Head.Ref)
 			}
 		}
-		refSpec := fmt.Sprintf("refs/pull/%s/head:%s", id, newBranchName)
 		newArgs = append(newArgs, newBranchName)
-		args.Before("git", "fetch", baseRemote.Name, refSpec)
+
+		ref := fmt.Sprintf("refs/pull/%s/head", id)
+		args.Before("git", "fetch", baseRemote.Name, fmt.Sprintf("%s:%s", ref, newBranchName))
+
+		remote := baseRemote.Name
+		mergeRef := ref
+		if pullRequest.MaintainerCanModify && pullRequest.Head.Repo != nil {
+			remote = pullRequest.Head.Repo.SshUrl
+			mergeRef = fmt.Sprintf("refs/heads/%s", pullRequest.Head.Ref)
+		}
+		args.Before("git", "config", fmt.Sprintf("branch.%s.remote", newBranchName), remote)
+		args.Before("git", "config", fmt.Sprintf("branch.%s.merge", newBranchName), mergeRef)
```

Comment:
Is it really necessary to set this configuration in the case when the PR is coming from the same repository, i.e. **not** from a maintainer-modifiable fork? I thought that [we already configure such branches properly](https://github.com/github/hub/blob/fe5b59b5489f2ac647c51914bf8e1b0e9993451f/features/checkout.feature#L97-L99).